src/ViewerDocument.tsx

```diff
@@ -0,0 +1,83 @@
+// eslint-disable-next-line no-use-before-define
+import React, { useState } from 'react'
+import ReactDOM from 'react-dom'
+import urlLogo from './assets/logo.png'
+import { ViewerState } from './viewer'
+import './style.css'
+
+export class ViewerDocument {
+  canvas: HTMLCanvasElement
+  logo: HTMLImageElement
+  link: HTMLAnchorElement
+  ui: HTMLDivElement
+  onProgress: (state: ViewerState) => void
+
+  constructor (settings: any, stateChangeEventName: string) {
+    // Get or Add Canvas
+    let canvas = document.getElementById(settings.canvasId)
+    if (!canvas) {
+      canvas = document.createElement('canvas')
+      document.body.appendChild(canvas)
+    }
+    this.canvas = canvas as HTMLCanvasElement
+
+    const ui = document.createElement('div')
+    ui.className = 'vim-ui'
+    ui.style.position = 'fixed'
+    ui.style.width = '100%'
+    ui.style.height = '100%'
+    ui.style.pointerEvents = 'none'
+    document.body.prepend(ui)
+    this.ui = ui
+
+    // Add Vim logo
+    const logo = document.createElement('img')
+    logo.src = urlLogo
+    logo.style.position = 'fixed'
+    logo.style.top = '16px'
+    logo.style.left = '16px'
+    logo.height = 48
+    logo.width = 128
+    this.logo = logo
+
+    // Add logo as link
+    const link = document.createElement('a')
+    link.href = 'https://vimaec.com'
+    link.appendChild(this.logo)
+    document.body.prepend(link)
+    this.link = link
+
+    ReactDOM.render(<VimUI eventName={stateChangeEventName} />, this.ui)
+  }
+}
+
+function VimUI (props: { eventName: string }) {
```

Comment:
I believe all of the HTML and CSS for loading and should be set-up in the host HTML page, and passed into the viewer.  I don't think the viewer should be bringing in React. Let's talk.